lib/linter/linter.js

```diff
@@ -456,10 +460,7 @@ function normalizeEcmaVersion(parser, ecmaVersion) {
     return ecmaVersion >= 2015 ? ecmaVersion - 2009 : ecmaVersion;
 }
 
-const eslintEnvPatterns = [
-    /\/\*\s*eslint-env\s(.+?)\*\//gsu, // block comments
-    /\/\/[^\S\r\n\u2028\u2029]*eslint-env[^\S\r\n\u2028\u2029](.+)/gu // line comments
-];
+const eslintEnvPattern = /\/\*\s*eslint-env\s(.+?)(?:\*\/|$)/gsu;
```

Comment:
I'm pretty sure this is a codeql false positive now. If the string starts with `'/*eslint-env '` and has some characters after that, this new regex must match something without ever looking for `'/*eslint-env '` again.

I just tested with 100,000 repetitions and it's working instantly (whereas the old regex hangs).

Reply:
marked as false positive.